### ISSUE TITLE
Review lab 2

### ISSUE THREAD

**Issue by Eliafontana** [b3xul/Computational_Intelligence#2]
#  Review lab 2
Overall the code is well written, below some comments and suggestions, I hope they can be useful:
* The readme is synthetic but well explain the whole process. Maybe you could add something about the data structure used and comments the results obtained. 
* The initialization of the population is remarkable. 
* About the fitness, I really like the way in which you used the counter module, I will probably steal it. However, in my opinion, from the comments it isn't so clear what the second term of the fitness means. It is the total number of elements considering also repetition, but total lenght, as you wrote, doesn't fully explain the value. 
* Since you are using crossover when you already have a complete solution, you can change the mutation rate over time in order to do more crossover operations in last generations. You could also consider adding a different crossover function, for first generations, that not necessary needed the constrain of full coverage. 

In conclusion I think you did a great job. Well done!
